src/queue.js

```diff
@@ -1,4 +1,21 @@
 class Queue {
+  constructor() {
+    this.queue = [];
+    this.size = this.queue.length;
+  }
+  enqueue(x) {
+    this.queue.unshift(x);
+    this.size = this.size + 1;
+  }
+  dequeue() {
+    if (this.size > 0) {
+      this.size = this.size - 1;
+    }
+    return this.queue.pop();
+  }
+  size() {
```

Comment:
Try to be more descriptive with this method name. Since it is setting `this.size` to 0, maybe name it `resetSize` or something like that.